script.js

```diff
@@ -0,0 +1,98 @@
+// // -----------------Storing information in Js object --------------------------------
+// const project = {
+//   name: 'portfolio',
+//   description:
+//     'Personal portfolio that contains project information, skills, and contact',
+//   technologies: ['HTML', 'CSS', 'JavaScript', 'Bootstrap'],
+//   liveVersion: 'https://ridabensalem.github.io/',
+//   sourceCode: 'https://github.com/ridabensalem/ridabensalem.github.io',
+// };
+//  -----------------Handiling the pop up functionality----------------------------
+const buttonPopup = document.querySelector('#pop-btn');
+const modalPopup = document.querySelector('.block-popup');
+const closeButton = document.querySelector('#close-btn');
+function Display() {
+  modalPopup.style.display = 'block';
+}
+function closePopup() {
+  modalPopup.style.display = 'none';
+}
+buttonPopup.addEventListener('click', Display);
+closeButton.addEventListener('click', closePopup);
+// ---------------------------Contact form validation---------------------
+const formContact = document.getElementById('form');
+
+function validateEmail() {
+  const email = document.getElementById('email').value;
+  const error = document.getElementById('errorMessage');
+  if (email.toLowerCase() !== email) {
+    error.innerHTML = 'Please enter your email in lowercase';
+    error.style.display = 'block';
+    error.style.color = '#FFFF';
+    error.style.fontSize = 'larger';
+    return false;
+  }
+  document.getElementById('errorMessage').innerHTML = '';
+  return true;
+}
+
+formContact.addEventListener('submit', validateEmail);
+
+// -----------------------Local storage --------------------
+window.onload = function storage() {
+  const fullName = document.getElementById('fullname');
+  const email = document.getElementById('email');
+  const textMessage = document.getElementById('message');
+
+  // retrieve data from local storage
+  fullName.value = localStorage.getItem('fullName');
+  email.value = localStorage.getItem('Email');
+  textMessage.value = localStorage.getItem('textMessage');
+
+  function saveToLocalStorage() {
+    // store data when user enters a new data
+    localStorage.setItem('fullName', fullName.value);
+    localStorage.setItem('Email', email.value);
+    localStorage.setItem('textMessage', textMessage.value);
+  }
+
+  // store data when user changes an existing data
+  fullName.addEventListener('input', saveToLocalStorage);
+  email.addEventListener('input', saveToLocalStorage);
+  textMessage.addEventListener('input', saveToLocalStorage);
+};
```

Comment:
- For some reason, data input to email and text message aren't saved on the local storage. But that of fullName saves. Can we fix this?? 🤗 

Here is what happens when I refresh the page after adding some data 👇🏽 

<img width="1376" alt="Screenshot 2023-01-06 at 10 19 30" src="https://user-images.githubusercontent.com/99246359/211043711-7b8aecab-6276-42a3-b270-2e5c9d716130.png">

Email and text message appear as empty strings.